contracts/extensions/VotingReputation.sol

```diff
@@ -0,0 +1,817 @@
+    require(pollState != PollState.Closed, "voting-rep-poll-escalation-window-open");
+    require(pollState != PollState.Executed, "voting-rep-poll-already-executed");
+    require(pollState == PollState.Executable, "voting-rep-poll-not-executable");
+
+    poll.executed = true;
+
+    bool canExecute = (
+      poll.stakes[NAY] <= poll.stakes[YAY] &&
+      poll.votes[NAY] <= poll.votes[YAY]
+    );
+
+    if (getSig(poll.action) == CHANGE_FUNCTION) {
+      bytes32 structHash = hashExpenditureStruct(poll.action);
+      expenditurePollCounts[structHash] = sub(expenditurePollCounts[structHash], 1);
+
+      // Release the claimDelay if this is the last active poll
+      if (expenditurePollCounts[structHash] == 0) {
+        bytes memory claimDelayAction = createClaimDelayAction(poll.action, 0);
+        executeCall(_pollId, claimDelayAction);
+      }
+
+      // Conditions:
+      //  - Yay side staked and nay side did not, and doman has sufficient vote power
+      //  - Both sides staked and yay side won, with sufficient vote power
+
+      uint256 votePower;
+      if (poll.stakes[NAY] < getRequiredStake(_pollId)) {
+        votePower = wmul(poll.skillRep, votePowerFraction);
+      } else {
+        votePower = poll.votes[YAY];
+      }
+
+      bytes32 slotHash = hashExpenditureSlot(poll.action);
+      if (expenditurePastPolls[slotHash] < votePower) {
+        expenditurePastPolls[slotHash] = votePower;
+        canExecute = canExecute && true;
+      } else {
+        canExecute = canExecute && false;
+      }
+    }
+
+    bool success;
+    if (canExecute) {
+      success = executeCall(_pollId, poll.action);
+    }
+
+    emit PollExecuted(_pollId, poll.action, success);
+  }
+
+  function claimReward(
+    uint256 _pollId,
+    uint256 _permissionDomainId, // For extension's arbitration permission
+    uint256 _childSkillIndex, // For extension's arbitration permission
+    address _user,
+    uint256 _vote
+  )
+    public
+  {
+    Poll storage poll = polls[_pollId];
+    require(
+      getPollState(_pollId) == PollState.Executed ||
+      getPollState(_pollId) == PollState.Failed,
+      "voting-rep-not-failed-or-executed"
+    );
+
+    // Calculate how much of the stake is left after voter compensation (>= 90%)
+    uint256 stake = stakes[_pollId][_user][_vote];
+    uint256 totalStake = add(poll.stakes[NAY], poll.stakes[YAY]);
+    uint256 rewardFraction = wdiv(poll.unpaidRewards, totalStake);
+    uint256 rewardStake = wmul(stake, rewardFraction);
+
+    delete stakes[_pollId][_user][_vote];
+
+    uint256 stakerReward;
+    uint256 repPenalty;
+
+    // Went to a vote, use vote to determine reward or penalty
+    if (
+      poll.stakes[NAY] == getRequiredStake(_pollId) &&
+      poll.stakes[YAY] == getRequiredStake(_pollId)
+    ) {
+      uint256 stakerVotes = poll.votes[_vote];
+      uint256 totalVotes = add(poll.votes[NAY], poll.votes[YAY]);
+      uint256 winPercent = wdiv(stakerVotes, totalVotes);
```

Comment:
Well... it's the percent of the vote which went to the staker's side... so it is the percent that the staker "won" -- not necessarily 51%+. I can "win" 30% of the vote and still lose, after all.

Reply:
I don't think this is a reasonable use of 'win'. You can talk about a losing side winning a certain number of votes, but that is not what this is. This is the fraction of the total turnout that this staker was. This number is not changed by the side anyone else votes on, just that they voted at all. And it's still not a percentage (see above).

Reply:
Not correct, I realize -- the naming of `stakerVotes` is misleading, as it is actually all the votes of the staker's side. So I think my argument stands? It's the % of the vote "won" by the staker's *side*. I can make it `winFraction` though since that's more in line with our convention for naming WAD values.